tests/test_device_attach_notification/src/test_device_attach_notification.xc

```diff
@@ -0,0 +1,191 @@
+// Copyright 2016-2025 XMOS LIMITED.
+// This Software is subject to the terms of the XMOS Public Licence: Version 1.
+
+#include <xs1.h>
+#include <print.h>
+#include <stdio.h>
+#include "xud.h"
+#include "platform.h"
+#include "xud_shared.h"
+
+#define XUD_EP_COUNT_OUT    2
+#define XUD_EP_COUNT_IN     2
+
+#ifndef PKT_LENGTH_START
+#define PKT_LENGTH_START    10
+#endif
+
+#ifndef PKT_LENGTH_END
+#define PKT_LENGTH_END      11
+#endif
+
+#ifndef TEST_EP_NUM
+#define TEST_EP_NUM         1
+#endif
+
+#ifndef XUD_TEST_SPEED
+#error XUD_TEST_SPEED not defined
+#endif
+
+/* Endpoint type tables */
+XUD_EpType epTypeTableOut[XUD_EP_COUNT_OUT] = {XUD_EPTYPE_CTL, XUD_EPTYPE_BUL | XUD_STATUS_ENABLE};
+XUD_EpType epTypeTableIn[XUD_EP_COUNT_IN] =   {XUD_EPTYPE_CTL, XUD_EPTYPE_BUL | XUD_STATUS_ENABLE};
+
+#pragma unsafe arrays
+int TestEpBusUpdate_Tx(chanend c_in, int epNum, unsigned start, unsigned end, t_runMode runMode)
+{
+    XUD_Result_t result;
+    XUD_BusSpeed_t busSpeed;
+    XUD_ep ep_in  = XUD_InitEp(c_in);
+    int i = 0;
+
+    unsigned char buffer[MAX_PKT_COUNT][1024];
+
+    int counter = 0;
+    int length = start;
+
+    set_core_fast_mode_on();
+
+    /* Prepare packets */
+    for(int i = 0; i <= (end - start); i++)
+    {
+        for(int j = 0; j < length; j++)
+        {
+            buffer[i][j] = counter++;
+        }
+        length++;
+    }
+
+#pragma loop unroll
+    length = start;
+    do
+    {
+        result = XUD_SetBuffer(ep_in, buffer[i], length);
+        if(result == XUD_RES_RST)
+        {
+            busSpeed = XUD_ResetEndpoint(ep_in, null);
+            assert(busSpeed == XUD_TEST_SPEED);
```

Comment:
Instead of asserting would it make more sense to return an appropriate failReason and have it printed in the TerminateFail() function to keep in line with how all the tests report failure and exit?

Reply:
Yes, fair point. Though im not sure failReason is being used properly at the moment.. I will take a look